org.scala-ide.sdt.core/src/org/scalaide/ui/internal/preferences/AutoEditsPreferencePage.scala

```diff
@@ -0,0 +1,190 @@
+    table.setHeaderVisible(true)
+    table.setLinesVisible(true)
+
+    val tcl = new TableColumnLayout
+    tableComposite.setLayout(tcl)
+
+    viewer = new CheckboxTableViewer(table)
+    viewer.setContentProvider(ContentProvider)
+    viewer.addSelectionChangedListener { e: SelectionChangedEvent =>
+      table.getSelection.headOption foreach { item ⇒
+        selectAutoEdit(item.getData().asInstanceOf[AutoEditSetting])
+      }
+    }
+    viewer.addCheckStateListener { e: CheckStateChangedEvent =>
+      toggleAutoEdit(e.getElement().asInstanceOf[AutoEditSetting])
+    }
+
+    val columnEnabled = new TableViewerColumn(viewer, SWT.NONE)
+    columnEnabled.getColumn().setText("Name")
+    columnEnabled onLabelUpdate { _.asInstanceOf[AutoEditSetting].name }
+    tcl.setColumnData(columnEnabled.getColumn(), new ColumnWeightData(1, true))
+
+    viewer.setInput(settings.sortBy(_.name))
+    viewer.setAllChecked(false)
+    viewer.setCheckedElements(settings.filter(isEnabled).asInstanceOf[Array[AnyRef]])
+
+    mkLabel(base, "Configuration:", columnSize = 2)
+    configurationArea = mkTextArea(base, lineHeight = 3, columnSize = 2)
+    configurationArea.addModifyListener { e: ModifyEvent ⇒
+
+      /**
+       * Checks if the following properties are hold:
+       * - each line contains one key-value pair
+       * - a key-value pair is separated by a '=' sign
+       * - no value needs to be specified, in this case the value is treated as
+       *   the value 'true'
+       */
+      def isValid(str: String): Boolean = {
+        str.isEmpty() || Commons.split(str, '\n').forall { line ⇒
+          Commons.split(line, '=') match {
+            case Seq(k, v) if k.nonEmpty ⇒ true
+            case Seq(k) if k.nonEmpty ⇒ true
+            case _ ⇒ false
+          }
+        }
+      }
+
+      val text = configurationArea.getText()
+      if (!isValid(text)) {
+        setValid(false)
+        setErrorMessage("Invalid configuration")
+      }
+      else {
+        setValid(true)
+        setErrorMessage(null)
+
+        viewer.getSelection() match {
+          case s: StructuredSelection ⇒
+            Option(s.getFirstElement()).map(_.asInstanceOf[AutoEditSetting]) foreach { aes ⇒
+              val existingConfig = autoEditConfig(aes)
+
+              configurations = configurations.filter(_._1 != aes)
+              if (existingConfig != text)
+                configurations += (aes → text)
+            }
+        }
+      }
+    }
+
+    mkLabel(base, "Description:", columnSize = 2)
+    descriptionArea = mkTextArea(base, lineHeight = 6, columnSize = 2, style = SWT.READ_ONLY)
+
+    base
+  }
+
+  override def init(workbench: IWorkbench): Unit = ()
+
+  override def performOk(): Boolean = {
+    changes foreach { autoEdit =>
+      val previousValue = prefStore.getBoolean(autoEdit.id)
+      prefStore.setValue(autoEdit.id, !previousValue)
```

Comment:
You should always save the state in the control. No need to invert the existing boolean value, it's easy to get out of sync, and even if you think that shouldn't happen, it's still unnecessarily stateful.